velox/vector/BaseVector.h

```diff
@@ -756,6 +756,11 @@ class BaseVector {
     memoDisabled_ = true;
   }
 
+  /// Used to check internal state of a vector like sizes of the buffers,
+  /// enclosed child vectors, values in indices. Currently, its only used in
+  /// debug builds to check the result of expressions and some interim results.
+  virtual void checkInternalState() const;
```

Comment:
I would vote for `validate()` if you consider a change for the function name.  Also it would be nice to have a `ValidateOptions` to be passed around to control different behaviors (most importantly adding different behaviors in the future without too much work):

```c++
struct ValidateOptions {
  bool loadLazy;
  std::function<void(BaseVector&)> callback;
};

virtual void validate(const ValidateOptions&);
```

Reply:
'validate' sounds like a good option, I'll make the change.
The idea for ValidateOptions is definitely useful however for the simplistic use case we have in this PR I would suggest we keep it minimalistic as we can always add more utilitarian functionality once we have a use case for it. What do you think?

Reply:
The idea is that we add the struct to the function signature now so that in the future people only need to add a field to the struct instead of updating the function with a million parameters.

Reply:
@Yuhta makes sense, added them. Please take another look when you get a chance